### PR TITLE
Write Plasma construction

### PR DESCRIPTION
**Pull request by syuhei176**

### CODE REVIEW

File: articles/OVM_architecture.md [hunk 1]
```diff
@@ -63,16 +63,91 @@ Table of contents:
...
+```
+plasma_exit(block_number, state_update) :=
+And(
+  IncludedAtBlock(block_number, state_update),
+  Not(IsDeprecated(state_update)),
+)
+```
+
+
+To exit state from Plasma, users should claim both checkpoint and exit claims.
```
Comment (yuriko627) [inline articles/OVM_architecture.md]: Maybe you should state this in line 70? After saying 'For simple Plasma cash, we require two claims'

File: articles/OVM_architecture.md [hunk 2]
```diff
@@ -63,16 +63,91 @@ Table of contents:
...
+      Or(
+        And(
+          IncludedAtBlock(state_update),
+          IsDeprecated(state_update)
+        ),
+        ExcludedAtBlock(state_update),
+      )
+```
+
+> For all b less than block number and for all state update included in block b have been excluded or, included and deprecated.
```
Comment (yuriko627) [inline articles/OVM_architecture.md]: For all state updates with a block number less than the current block number, there does not exist the same coin range AND excluded or included then deprecated.

File: articles/OVM_architecture.md [hunk 3]
```diff
@@ -63,16 +63,91 @@ Table of contents:
...
+
+
+```
+plasma_exit(block_number, state_update) :=
+And(
+  IncludedAtBlock(block_number, state_update),
+  Not(IsDeprecated(state_update)),
+)
+```
+
```
Comment (yuriko627) [inline articles/OVM_architecture.md]: > There exists a state update that is included in the block, and for all state updates with a higher nonce than this exit state update, there does not exist a deprecation

File: articles/OVM_architecture.md (hunk 2 above)
Comment (syuhei176) [reply articles/OVM_architecture.md]: I've changed checkpoint claim little bit. I remove excludedAtBlock. So I changed also natural language.

```
For all state updates with a block number less than the current block number, state updates should be included and deprecated.
```

Comment (yuriko627) [reply articles/OVM_architecture.md]: I thought you had to state the claim in the format of "For all xx, there does not exist yyy" in order to make it possible to make a contradiction claim. If you prove that there in fact DOES exist yyy, then someone can challenge the statement submitting the proof of existing yyy. (remember ugly cat's example on Takamichi's slide?)

Comment (yuriko627) [reply articles/OVM_architecture.md]: or maybe you can write it out "For all state updates with a block number less than the current block number, there exists inclusion proof and deprecation proof."

File: articles/OVM_architecture.md [hunk 4]
```diff
@@ -63,16 +63,90 @@ Table of contents:
...
+When challengers challenge exit.
+
+For challenge-not-inclusion, a challenger can prove undecided contradiction of `IncludedAtBlock(block_number, state_update)` in exit claim showing `Not(IncludedAtBlock(block_number, state_update))`. And also the challenger can prove deprecation showing `IsDeprecated(state_update)`. This is contradiction of exit claim.
+
+Challenge for checkpoint(invalid history challenge) is little bit complex, but OVM can be mitigating difficulty. A challenger can prove invalid history showing `And(IncludedAtBlock(state_update), Not(IsDeprecated(state_update)))`. In this case, exitor should show inclusion of challenging state_update.
+
+
 ```
 - what can application developers build on top of this system?
```
Comment (yuriko627) [inline articles/OVM_architecture.md]: Are you gonna mention how to build an app in this blog post? I thought that would be the next one.

Comment (syuhei176) [reply articles/OVM_architecture.md]: Yes, it can be removed

Comment (yuriko627) [reply articles/OVM_architecture.md]: let's save it for later!

Comment (syuhei176) [reply articles/OVM_architecture.md]: removed

**Pull request closed without merging**